### PR TITLE
Moe/ipfs uploader

### PR DESCRIPTION
**Pull request by Moejoe90**
This PR passes with all required specification
- More work needs to be done with cleaning code in terms File reading and error handling
- Only tested with private local node but should work 
`yarn uploadAgreements "ipfs.neustg.net" "./legal/NEUMARK TOKEN HOLDER AGREEMENT.out.html"`
or default
`yarn uploadAgreements`

### CODE REVIEW

File: scripts/uploadAgreements.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+/* eslint-disable no-console */
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+const ipfsAPI = require("ipfs-api");
+const fs = require("fs");
+const path = require("path");
+
+const isFilePinned = async (ipfs, hash) => {
+  const matchedHashes = (await ipfs.pin.ls()).filter(response => {
```
Comment (rudolfix) [inline scripts/uploadAgreements.js]:
```
{
 +    if (response.hash === hash) return true;
 +    return false;
 +  }
```
may be replaced with single `response.hash === hash`

File: scripts/uploadAgreements.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        file: fs.readFileSync(filePath)
+      };
+    } catch (e) {
+      throw new Error(`Can't read file "${filePath}"`);
+    }
+  });
+
+// TODO: handle diffrent ports for ipfs
+const main = async ([ipfsNodeAddress, ...paths]) => {
+  const defaultDirpath = path.join(__dirname, "..", "legal");
```
Comment (rudolfix) [inline scripts/uploadAgreements.js]: rename to `defaultDirPath`

File: scripts/uploadAgreements.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  const filePaths = [];
```
Comment (rudolfix) [inline scripts/uploadAgreements.js]: fileNames

File: scripts/uploadAgreements.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  const defaultfilePaths = [
```
Comment (rudolfix) [inline scripts/uploadAgreements.js]: `defaultFileNames` paths are not names. also mind capitalization

File: scripts/uploadAgreements.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    "NEUMARK TOKEN HOLDER AGREEMENT.out.html",
+    "RESERVATION AGREEMENT.out.html"
+  ];
+  try {
+    if (!paths) throw new Error("Please give ipfs node");
```
Comment (rudolfix) [inline scripts/uploadAgreements.js]: huh? you should check `ipfsNodeAddress` for being empty here?

File: scripts/uploadAgreements.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        name: path.parse(filePath).base,
+        file: fs.readFileSync(filePath)
+      };
+    } catch (e) {
+      throw new Error(`Can't read file "${filePath}"`);
+    }
+  });
+
+// TODO: handle diffrent ports for ipfs
+const main = async ([ipfsNodeAddress, ...paths]) => {
```
Comment (rudolfix) [inline scripts/uploadAgreements.js]: are those `paths` or `fileNames`?

File: scripts/uploadAgreements.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+const isFilePinned = async (ipfs, hash) => {
+  const matchedHashes = (await ipfs.pin.ls()).filter(response => {
+    if (response.hash === hash) return true;
+    return false;
+  });
+  return matchedHashes.length > 0;
+};
+
+const addFiletoIpfs = async (ipfs, file, name) => {
+  try {
```
Comment (rudolfix) [inline scripts/uploadAgreements.js]: do we need try...catch here? just let exception go. you will catch it later

File: scripts/uploadAgreements.js (hunk 5 above)
Comment (Moejoe90) [reply scripts/uploadAgreements.js]: good point i missed this one when refactoring code

File: scripts/uploadAgreements.js (hunk 1 above)
Comment (Moejoe90) [reply scripts/uploadAgreements.js]: Done

File: scripts/uploadAgreements.js (hunk 7 above)
Comment (Moejoe90) [reply scripts/uploadAgreements.js]: done

File: scripts/uploadAgreements.js (hunk 4 above)
Comment (Moejoe90) [reply scripts/uploadAgreements.js]: Done

File: scripts/uploadAgreements.js (hunk 6 above)
Comment (Moejoe90) [reply scripts/uploadAgreements.js]: These are indeed paths

File: scripts/uploadAgreements.js (hunk 2 above)
Comment (Moejoe90) [reply scripts/uploadAgreements.js]: Done

File: scripts/uploadAgreements.js (hunk 3 above)
Comment (Moejoe90) [reply scripts/uploadAgreements.js]: This is actually file paths

**Pull request merged**